pyiceberg/types.py

```diff
@@ -703,6 +718,44 @@ class TimestamptzType(PrimitiveType):
     root: Literal["timestamptz"] = Field(default="timestamptz")
 
 
+class TimestampNanoType(PrimitiveType):
+    """A TimestampNano data type in Iceberg can be represented using an instance of this class.
+
+    TimestampNanos in Iceberg have nanosecond precision and include a date and a time of day without a timezone.
+
+    Example:
+        >>> column_foo = TimestampNanoType()
+        >>> isinstance(column_foo, TimestampNanoType)
+        True
+        >>> column_foo
+        TimestampNanoType()
+    """
+
+    root: Literal["timestamp_ns"] = Field(default="timestamp_ns")
+
+    def minimum_format_version(self) -> TableVersion:
+        return TableVersion.THREE
```

Comment:
UnknownType was just added in #1681 lets make sure this is properly set there too

Reply:
great suggestion @kevinjqliu thank you!